llvm/test/CodeGen/AMDGPU/llvm.amdgcn.update.dpp.gfx90a.ll

```diff
@@ -0,0 +1,258 @@
+  store <8 x half> %tmp0, ptr addrspace(1) %out
+  ret void
+}
+
+; GCN-LABEL: update_dpp_p3_poison:
+;
+; GCN:        v_mov_b32_dpp v0, v0 row_newbcast:1 row_mask:0x1 bank_mask:0x1
+;
+; GCN:        global_store_dword v[{{[0-9:]+}}], v0, off
+
+define amdgpu_ps void @update_dpp_p3_poison(ptr addrspace(3) %in, ptr addrspace(1) %out) {
+  %tmp0 = call ptr addrspace(3) @llvm.amdgcn.update.dpp.p3(ptr addrspace(3) poison, ptr addrspace(3) %in, i32 337, i32 1, i32 1, i1 0)
+  store ptr addrspace(3) %tmp0, ptr addrspace(1) %out
+  ret void
+}
+
+; GCN-LABEL: update_dpp_p0_poison:
+;
+; GFX90A-DAG:    v_mov_b32_dpp v0, v0 row_newbcast:1 row_mask:0x1 bank_mask:0x1
+; GFX90A-DAG:    v_mov_b32_dpp v1, v1 row_newbcast:1 row_mask:0x1 bank_mask:0x1
+;
+; GFX942-DAG:    v_mov_b64_dpp v[0:1], v[0:1] row_newbcast:1 row_mask:0x1 bank_mask:0x1
+;
+; GCN:            global_store_dwordx2 v[2:3], v[0:1], off
+
+define amdgpu_ps void @update_dpp_p0_poison(ptr %in, ptr addrspace(1) %out) {
+  %tmp0 = call ptr @llvm.amdgcn.update.dpp.p0(ptr poison, ptr %in, i32 337, i32 1, i32 1, i1 0)
+  store ptr %tmp0, ptr addrspace(1) %out
+  ret void
+}
+
+; GCN-LABEL: update_dpp_i64_unsupported_dpp64_op:
+;
+; GCN-DAG:    v_mov_b32_dpp v3, v1 quad_perm:[1,0,0,0] row_mask:0x1 bank_mask:0x1
+; GCN-DAG:    v_mov_b32_dpp v2, v0 quad_perm:[1,0,0,0] row_mask:0x1 bank_mask:0x1
+;
+; GCN:        global_store_dwordx2 v[4:5], v[2:3], off
+
+define amdgpu_ps void @update_dpp_i64_unsupported_dpp64_op(i64 %in, i64 %old, ptr addrspace(1) %out) {
+  %tmp0 = call i64 @llvm.amdgcn.update.dpp.i64(i64 %old, i64 %in, i32 1, i32 1, i32 1, i1 0)
+  store i64 %tmp0, ptr addrspace(1) %out
+  ret void
+}
+
+; GCN-LABEL: update_dpp_i16:
+;
+; GCN:        v_mov_b32_dpp v1, v0 row_newbcast:1 row_mask:0x1 bank_mask:0x1
+;
+; GCN:        global_store_short v[2:3], v1, off
+
+define amdgpu_ps void @update_dpp_i16(i16 %in, i16 %old, ptr addrspace(1) %out) {
+  %tmp0 = call i16 @llvm.amdgcn.update.dpp.i16(i16 %old, i16 %in, i32 337, i32 1, i32 1, i1 0)
+  store i16 %tmp0, ptr addrspace(1) %out
+  ret void
+}
+
+; GCN-LABEL: update_dpp_half:
+;
+; GCN-DAG:    v_mov_b32_dpp v1, v0 row_newbcast:1 row_mask:0x1 bank_mask:0x1
+;
+; GCN:        global_store_short v[2:3], v1, off
+
+define amdgpu_ps void @update_dpp_half(half %in, half %old, ptr addrspace(1) %out) {
+  %tmp0 = call half @llvm.amdgcn.update.dpp.f16(half %old, half %in, i32 337, i32 1, i32 1, i1 0)
+  store half %tmp0, ptr addrspace(1) %out
+  ret void
+}
+
+; GCN-LABEL: update_dpp_bfloat:
+;
+; GCN-DAG:    v_mov_b32_dpp v1, v0 row_newbcast:1 row_mask:0x1 bank_mask:0x1
+;
+; GCN:        global_store_short v[2:3], v1, off
+
+define amdgpu_ps void @update_dpp_bfloat(bfloat %in, bfloat %old, ptr addrspace(1) %out) {
+  %tmp0 = call bfloat @llvm.amdgcn.update.dpp.bf16(bfloat %old, bfloat %in, i32 337, i32 1, i32 1, i1 0)
+  store bfloat %tmp0, ptr addrspace(1) %out
+  ret void
+}
```

Comment:
Test vector of pointers

Reply:
Added. Some vectors of pointers cannot be handled by global ISel even w/o the split, such as <2 x p3>, it just cannot select. But that is not a problem of this patch. SDag works on any combination though.